ShowBuildMenu.sh

```diff
@@ -0,0 +1,207 @@
+#!/bin/sh
+
+BUILD_PROJECT="./Tools/BuildTool/BuildTool.csproj"
+AVAILABLE_CONFIGURATIONS="available-test-configurations"
+CONFIG_NAME=""
+TEST_PLATFORM=""
+LIB_FILES=""
+LIB_FILES2=""
+CURRENT_CONFIGURATION="./current-test-configuration"
+BUILD_TOOL="dotnet run -p $BUILD_PROJECT -c Release --no-build -v q"
+OPTION=0
+
+buildDebug(){
+	eval "dotnet build ./src/NHibernate.sln"
+	echo "."
+	echo "Assuming the build succeeded, your results will be in the build folder."
+	echo "."
+	mainMenu
+}
+
+buildRelease(){
+	eval "dotnet build ./src/NHibernate.sln -c Release"
+	echo "."
+	echo "Assuming the build succeeded, your results will be in the build folder."
+	echo "."
+	mainMenu
+}
+
+testActivate(){
+	FILE_TEMP="folder.tmp"
+
+	$BUILD_TOOL pick-folder $AVAILABLE_CONFIGURATIONS $FILE_TEMP "Which test configuration should be activated?"
+
+	if [ -d $CURRENT_CONFIGURATION ]
+	then
+		rm -r $CURRENT_CONFIGURATION/
+	fi
+
+	CURRENT_FOLDER=$(pwd)
+	INFORMATION=$(cat $CURRENT_FOLDER/$FILE_TEMP)
+	cp -r $INFORMATION/ $CURRENT_CONFIGURATION
+
+	rm $FILE_TEMP
+
+	echo "Configuration activated."
+
+	mainMenu
+}
+
+testSetupGeneric() {
+	echo "Enter a name for your test configuration or press enter to use default name:"
+	read CFGNAME
+	if [ $CFGNAME = ""]
+	then
+		CFGNAME="$CONFIG_NAME-$TEST_PLATFORM"
+		echo $CFGNAME
+	fi
+
+	mkdir -p $AVAILABLE_CONFIGURATIONS/$CFGNAME
+
+	if [ ! "$LIB_FILES" ]
+	then
+		cp $LIB_FILES $AVAILABLE_CONFIGURATIONS/$CFGNAME
+
+		if [ ! "$LIB_FILES2" ]
+		then
+			cp $LIB_FILES2 $AVAILABLE_CONFIGURATIONS/$CFGNAME
+		fi
+	fi
+
+	cp "src/NHibernate.Config.Templates/$CONFIG_NAME.cfg.xml" "$AVAILABLE_CONFIGURATIONS/$CFGNAME/hibernate.cfg.xml"
+	echo "Done setting up files. Please edit the connection string in file:"
+	echo "$AVAILABLE_CONFIGURATIONS/$CFGNAME/hibernate.cfg.xml"
+	echo "When you're done, don't forget to activate the configuration through the menu."
+	mainMenu
+}
+
+testSetupSqlServer() {
+	CONFIG_NAME="MSSQL"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupFirebird() {
+	CONFIG_NAME="FireBird"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupSqlite() {
+	CONFIG_NAME="SQLite"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""e
+	testSetupGeneric
+}
+
+testSetupPostgresql() {
+	CONFIG_NAME="PostgreSQL"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupMysql() {
+	CONFIG_NAME="MySql"
+	TEST_PLATFORM="AnyCPU"
+	LIB_FILES=""
+	LIB_FILES2=""
+	testSetupGeneric
+}
+
+testSetupMenu() {
+	echo "A. Add a test configuration for SQL Server."
+	echo "B. Add a test configuration for Firebird."
+	echo "C. Add a test configuration for SQLite."
+	echo "D. Add a test configuration for PostgreSQL."
+	echo "F. Add a test configuration for MySql."
+	echo "."
```

Comment:
The prompt is still prompting for `E`, which will now just fully exit the menu. Better switch MySQL to `E` and renumber test 6 and 5 to 5 and 4.
(Contributing do not mention letters of sub-menus. It mentions only those of the main menu. So for sub-menu, no need to keep them in sync with the .bat IMO.)